src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFile.cs

```diff
@@ -128,7 +128,44 @@ public bool Equals(LockFile other)
                 && Libraries.OrderedEquals(other.Libraries, library => library.Name, StringComparer.OrdinalIgnoreCase)
                 && Targets.OrderedEquals(other.Targets, target => target.Name, StringComparer.Ordinal)
                 && PackageFolders.SequenceEqual(other.PackageFolders)
-                && EqualityUtility.EqualsWithNullCheck(PackageSpec, other.PackageSpec);
+                && EqualityUtility.EqualsWithNullCheck(PackageSpec, other.PackageSpec)
+                && LogsEqual(other.LogMessages);
+        }
+
+        private bool LogsEqual(IList<IAssetsLogMessage> otherLogMessages)
+        {
+            if(LogMessages == otherLogMessages)
```

Comment:
This would be a reference equals check? I'm not sure how List would handle this. I suggest removing this or using ReferenceEquals to make sure it doesn't do something unexpected